config/peerpods/podvm/libvirt-podvm-image-handler.sh

```diff
@@ -21,23 +21,75 @@ function verify_vars() {
 
     [[ -z "${ORG_ID}" ]] && error_exit "ORG_ID is not set"
     [[ -z "${ACTIVATION_KEY}" ]] && error_exit "ACTIVATION_KEY is not set"
-    [[ -z "${BASE_OS_VERSION}" ]] && error_exit "BASE_OS_VERSION is not set"
 
-    [[ -z "${PODVM_DISTRO}" ]] && error_exit "PODVM_DISTRO is not set"
+    if [[ "${IMAGE_TYPE}" == "operator-built" ]]; then
+        [[ -z "${BASE_OS_VERSION}" ]] && error_exit "BASE_OS_VERSION is not set"
 
-    [[ -z "${CAA_SRC}" ]] && error_exit "CAA_SRC is empty"
-    [[ -z "${CAA_REF}" ]] && error_exit "CAA_REF is empty"
+        [[ -z "${PODVM_DISTRO}" ]] && error_exit "PODVM_DISTRO is not set"
 
-    [[ -z "${REDHAT_OFFLINE_TOKEN}" ]] && error_exit "Redhat token is not set"
+        [[ -z "${CAA_SRC}" ]] && error_exit "CAA_SRC is empty"
+        [[ -z "${CAA_REF}" ]] && error_exit "CAA_REF is empty"
 
-    # Ensure booleans are set
-    [[ -z "${DOWNLOAD_SOURCES}" ]] && error_exit "DOWNLOAD_SOURCES is empty"
+        [[ -z "${REDHAT_OFFLINE_TOKEN}" ]] && error_exit "Redhat token is not set"
+
+        # Ensure booleans are set
+        [[ -z "${DOWNLOAD_SOURCES}" ]] && error_exit "DOWNLOAD_SOURCES is empty"
+    fi
 }
 
+# Function to create the libvirt image based on the type.
+function create_libvirt_image() {
+    # If the value of the PODVM_IMAGE_URI is empty, then build the image from scratch else use the prebuilt artifact.
+    if [[ "${IMAGE_TYPE}" == "operator-built" ]]; then
+        create_libvirt_image_from_scratch
+    elif [[ "${IMAGE_TYPE}" == "pre-built" ]]; then
+        create_libvirt_image_from_prebuilt_artifact
+    fi
 
+}
 
-function create_libvirt_image() {
-    echo "Creating libvirt qcow2 image"
+# Function to create the libvirt image from a prebuilt artifact.
+function create_libvirt_image_from_prebuilt_artifact() {
+    echo "Pulling qcow2 image for libvirt provider from the provided path"
+    PODVM_IMAGE_SRC="/tmp/image"
+    PODVM_DESTINATION_PATH="/podvm"
+    PODVM_ROOTFS_PATH="/rootfs"
```

Comment:
if this fixed, maybe just use it in the forming or remove the podvm_prefix as i don't find it directly related to the podvm itself but to the container extraction in general

Reply:
I think only the `PODVM_DESTINATION_PATH` is being used multiple times, rest all are one time used, will clean them up.